File: buffer_overflow.cpp
/*
CS638
Exercise on Buffer Overflow (Part A and B)

@author Vishnu Sai Rao Suresh Lokhande (lokhande@cs.wisc.edu)
*/



#include <iostream>
#include <string>
#include <cstring>
#include <exception>

using namespace std;

int part_a() {
  int a[4];
  try {
    a[0] = 20;
    a[100] = 23; // Assigning element out of range
  } catch (exception& e) {
    cout << e.what() << "\n";
  }
  return 0;
}

int part_b() {
  string str1 = "Hello";
  try {
    cout << str1[2] << "\n";
    cout << str1[10000] << "\n"; // Accessing element out of range
  } catch (exception& e) {
    cout << e.what() << "\n";
  }
    
  return 0;
}

int main() {
  part_a();
  part_b();

  return 0;
}

File: buffer_overflow.py
'''
CS638
Exercise on Buffer Overflow (Part D)

@author Vishnu Sai Rao Suresh Lokhande (lokhande@cs.wisc.edu)

'''


def main():
    a = [0]*10;
    try:
        print(a[1]);
        print(a[1000]); # accessing element out of range
    except:
        print("EXCEPTION");

if __name__ == '__main__':
    main()
